shared/search/result-row/container.js

```diff
@@ -1,38 +1,37 @@
 // @flow
 import SearchResultRow from '.'
-import {Map} from 'immutable'
 import {userIsActiveInTeamHelper} from '../../constants/teams'
-import {followStateHelper} from '../../constants/search'
+import {followStateHelper, makeSearchResult} from '../../constants/search'
 import {type SearchResultId} from '../../constants/types/search'
 import {connect, type TypedState, setDisplayName, compose} from '../../util/container'
 
-type OwnProps = {
+export type OwnProps = {
   disableIfInTeamName: ?string,
   id: SearchResultId,
+  selected: boolean,
   onClick: () => void,
   onMouseOver?: () => void,
   onShowTracker?: () => void,
 }
 
-const mapStateToProps = (
-  state: TypedState,
-  {disableIfInTeamName, id, onClick, onMouseOver, onShowTracker}: OwnProps
-) => {
-  const result: any = state.entities.getIn(['search', 'searchResults', id], Map()).toObject()
+const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
+  const result = state.entities.search.searchResults.get(ownProps.id, makeSearchResult()).toObject()
```

Comment:
i know this is old code but we should never do this. do this in mergeprops

Reply:
Done